### PR TITLE
Mask to simple fit

### PR DESCRIPTION
**Pull request by j-ostmeyer**
... and simplify code a bit.

### CODE REVIEW

**Comment by j-ostmeyer** [comment]
See conversation in #217 .

**Comment by j-ostmeyer** [comment]
What is the status of priors in simple.nlsfit? One should either remove the stop-statement (if implemented), or the code segments with priors (if not), or implement it.

File: R/removeTemporal.cf.R [hunk 1]
```diff
@@ -378,10 +380,10 @@ weight.cf <- function (cf, energy_difference_val, energy_difference_boot,
 #' @inheritParams weight.cf
 weight_shift_reweight.cf <- function (cf, energy_difference_val, energy_difference_boot, cosh_factor) {
   cf <- weight.cf(cf, energy_difference_val, energy_difference_boot,
-                  cosh_factor, 0, FALSE)
+                  cosh_factor, 0, TRUE)
```
Comment (kostrzewa) [inline R/removeTemporal.cf.R]: is it normal that this PR touches this?

Comment (kostrzewa) [reply R/removeTemporal.cf.R]: ah, this is outdated

**Comment by j-ostmeyer** [comment]
This breaks down somewhere in Hankel. Is the problem solved by now and simply not merged into this branch, or it is something more complicated?

**Comment by urbach** [comment]
no, this is still the `multi_particle_fit_test.Rmd` vignette, not hankel.

**Comment by urbach** [comment]
this is a crude fix, by not evaluating the problematic chunks. This needs to be remembered, of course, for later...

**Comment by urbach** [comment]
now package `stringi` is the problem, like for hankel... Something must have changed...

**Comment by urbach** [comment]
> This breaks down somewhere in Hankel. Is the problem solved by now and simply not merged into this branch, or it is something more complicated?

its solved in the hankel branch, though not realted to the hankel
method. For some reason, bibtex does not work anymore, which I also do
not understand at all. Still worked for me on Monday...

**Comment by j-ostmeyer** [comment]
No idea, what happens there. Could you be a bit more specific, please? My fits work.

**Comment by j-ostmeyer** [comment]
I hope that everything is fine now. Just so you know: The code never gave correct results for xy-errors with mask or priors. It simply did not crash out of courtesy of R. Vectors of different lengths can be multiplied...

**Comment by martin-ueding** [comment]
Oh wow, good that you found these issues! Luckily I only use the combinations (y-errors, covariance, priors, mask) and (xy-errors, covariance) but not xy-errors with either mask or covariance.

**Comment by martin-ueding** [comment]
I have just tested this on my 3pi Lüscher Analysis and it produces the exact same results.

**Pull request merged**